### PR TITLE
CI: update codecov

### PR DESCRIPTION
**Pull request by tacaswell**
codecov has changed how they discover the yaml file (it apparently
used to search whole repo, now only looks at top level).

Try to explicitly tell it where to look and to add azure to the set of
things that uploads.

### CODE REVIEW

**Comment by tacaswell** [comment]
So codecov upload does not work on windows _and_ I did not sush the bot...

**Comment by tacaswell** [comment]
ok, this is now pushing to codecov from azure on linux + osx, one more try at the win version.

**Comment by tacaswell** [comment]
Added the no-comment flag to the "team yaml" https://codecov.io/account/gh/matplotlib/yaml so hopefully that will stop the spam.  uploading is still not working in windows on azure :(

**Comment by tacaswell** [comment]
So this definitely makes it _better_. 

I think I want to squash this to one commit and remove the attempt at windows on azure first.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -101,6 +101,24 @@ steps:
...
+- bash: |
+    case "$(python -c 'import sys; print(sys.platform)')" in
+    linux)
+      bash <(curl -s https://codecov.io/bash)  -f "!*.gcov" -X gcov -y ci/codecov.yml
+      ;;
+    darwin)
+      bash <(curl -s https://codecov.io/bash) -f "!*.gcov"
+      ;;
+    win32)
+      choco install codecov
```
Comment (Kojoley) [inline azure-pipelines.yml]: Try `pip install codecov` like we do on Appveyor, choco seems to be installing `codecov-exe`

**Comment by Kojoley** [comment]
```
Error: Missing repository upload token
```

**Comment by tacaswell** [comment]
https://codecov.io/gh/matplotlib/matplotlib/commit/5658ca52c4f133013067e5b9a1a5b51f69d13e22/build <- it looks like now travis didn't get uploaded on that commit?!

https://developercommunity.visualstudio.com/idea/615016/integration-with-codecov-without-token-for-public.html <- suggests this (as of last summer) was still an outstanding problem

https://github.com/codecov/codecov-exe <- this suggests that it codecov-exe only needs a token for private repos 

https://github.com/codecov/codecov-python <- azure is not in this list

https://devblogs.microsoft.com/devops/uploading-to-codecov-just-got-easier/ <- this (from MSFT!) only works on linux

**Comment by Kojoley** [comment]
> codecov.io/gh/matplotlib/matplotlib/commit/5658ca52c4f133013067e5b9a1a5b51f69d13e22/build <- it looks like now travis didn't get uploaded on that commit?!

```
$ bash <(curl -s https://codecov.io/bash) -f "!*.gcov" -X gcov -e TRAVIS_PYTHON_VERSION -y ci/codecov.yml|| echo "Codecov did not collect coverage reports"
/dev/fd/63: line 1: html: No such file or directory
/dev/fd/63: line 2: syntax error near unexpected token `<'
'
Codecov did not collect coverage reports
```

The master branch has the same problem. The previous (successful) commit (in this PR) was ~at the same time as https://github.com/codecov/codecov-bash update.

**Comment by Kojoley** [comment]
Now all Azure jobs uploading coverage reports, but I do not understand what Codecov is complaining about.

**Comment by tacaswell** [comment]
Weird, I tried just using the `bash <(curl ...)` method first and it did not work on windows...I assume there were upstream changes?

It looks like codecov has become confused with how it is computing hit changes (there is at least one line that is saying the `def ...` line is not hit, but the body of the same function _is_.

I'm going to squash-merge this.  Thank you for your help on this @Kojoley !

**Pull request merged**